hi-ml/src/health_ml/deep_learning_config.py

```diff
@@ -0,0 +1,722 @@
+#  ------------------------------------------------------------------------------------------
+#  Copyright (c) Microsoft Corporation. All rights reserved.
+#  Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+#  ------------------------------------------------------------------------------------------
+from __future__ import annotations
+
+import logging
+from enum import Enum, unique
+from pathlib import Path
+from typing import Any, Dict, List, Optional
+
+import param
+from pandas import DataFrame
+from param import Parameterized
+
+from health_azure.utils import RUN_CONTEXT, GenericConfig, PathOrString, is_offline_run_context, T
+
+from health_ml.utils import fixed_paths
+from health_ml.utils.common_utils import (create_unique_timestamp_id, DATASET_CSV_FILE_NAME,
+                                          DEFAULT_CROSS_VALIDATION_SPLIT_INDEX,
+                                          DEFAULT_AML_UPLOAD_DIR, DEFAULT_LOGS_DIR_NAME, is_windows,
+                                          ModelExecutionMode, ModelProcessing)
+from health_ml.utils.type_annotations import TupleFloat2
+
+
+@unique
+class LRWarmUpType(Enum):
+    """
+    Supported LR warm up types for model training
+    """
+    NoWarmUp = "NoWarmUp"
+    Linear = "Linear"
+
+
+@unique
+class LRSchedulerType(Enum):
+    """
+    Supported lr scheduler types for model training
+    """
+    Exponential = "Exponential"
+    Step = "Step"
+    Polynomial = "Polynomial"
+    Cosine = "Cosine"
+    MultiStep = "MultiStep"
+
+
+@unique
+class OptimizerType(Enum):
+    """
+    Supported optimizers for model training
+    """
+    Adam = "Adam"
+    AMSGrad = "AMSGrad"
+    SGD = "SGD"
+    RMSprop = "RMSprop"
+
+
+class DeepLearningFileSystemConfig(Parameterized):
```

Comment:
I probably introduced this class name, but I've now often thought it's weird. OutputFolderHandler?